packages/mdl-textfield/index.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import MDLComponent from 'mdl-base';
+
+import MDLTextfieldFoundation from './foundation';
+
+export {MDLTextfieldFoundation};
+
+const {cssClasses} = MDLTextfieldFoundation;
+
+let idCounter = 0;
+
+export default class MDLTextfield extends MDLComponent {
+  static buildDom({
+    id = `mdl-textfield-${++idCounter}`,
+    helptextId = '',
+    labelElement = null
+  } = {}) {
+    const root = document.createElement('div');
+    const {ROOT: CSS_ROOT} = cssClasses;
+    root.classList.add(CSS_ROOT);
+    root.innerHTML = `
+      <input type="text" class="${CSS_ROOT}__input" id="${id}"
+             ${helptextId ? ` aria-controls="${helptextId}" ` : ''}>
+    `;
+    if (labelElement) {
+      labelElement.classList.add(`${CSS_ROOT}__label`);
+      labelElement.setAttribute('for', id);
+      root.appendChild(labelElement);
+    }
+
+    return root;
+  }
+
+  static attachTo(root) {
+    return new MDLTextfield(root);
+  }
+
+  constructor() {
+    super(...arguments);
+    const input = this.input_;
+    this.helptextElement = input.hasAttribute('aria-controls') ?
+      document.getElementById(input.getAttribute('aria-controls')) : null;
+  }
+
+  initialSyncWithDom() {
+    this.disabled = this.input_.disabled;
+  }
+
+  get disabled() {
+    return this.foundation_.isDisabled();
+  }
+
+  set disabled(disabled) {
+    this.foundation_.setDisabled(disabled);
+  }
+
+  get input_() {
+    return this.root_.querySelector(`.${cssClasses.ROOT}__input`);
+  }
+
+  get label_() {
+    return this.root_.querySelector(`.${cssClasses.ROOT}__label`);
+  }
+
+  getDefaultFoundation() {
+    return new MDLTextfieldFoundation(Object.assign({
+      addClass: className => this.root_.classList.add(className),
+      removeClass: className => this.root_.classList.remove(className),
+      addClassToLabel: className => this.label_.classList.add(className),
+      removeClassFromLabel: className => this.label_.classList.remove(className)
+    }, this.getInputAdapterMethods_(), this.getHelptextAdapterMethods_()));
+  }
+
+  getInputAdapterMethods_() {
```

Comment:
Simply for style, so that the `getDefaultAdapter()` method does not include a gigantic object.